src/supplemental/nanolib/parquet/parquet.cc

```diff
@@ -490,6 +484,46 @@ parquet_write(
 			goto again;
 	}
 
+	char md5_buffer[33];
+	int ret = Compute_file_md5(filename, md5_buffer);
+	if (ret != 0) {
+		log_error("Failed to calculate md5sum");
+		free(filename);
+		return -1;
+	}
+
+	char *md5_file_name = (char *)malloc(strlen(filename) + strlen("_") + strlen(md5_buffer) + 2);
+	if (md5_file_name == NULL) {
+		log_error("Failed to allocate memory for file name.");
+		free(filename);
+		return -1;
+	}
+
+	strncpy(md5_file_name, filename, strlen(conf->dir) + strlen(conf->file_name_prefix) + 1);
+	md5_file_name[strlen(conf->dir) + strlen(conf->file_name_prefix) + 1] = '\0';
+
+	strcat(md5_file_name, "_");
+	strcat(md5_file_name, md5_buffer);
+	strcat(md5_file_name, filename + strlen(conf->dir) + strlen(conf->file_name_prefix) + 1);
+
+	ret = rename(filename, md5_file_name);
+	if (ret != 0) {
+		log_error("Failed to rename file %s to %s errno: %d", filename, md5_file_name, errno);
+		free(filename);
+		free(md5_file_name);
+		return -1;
+	}
+
+	free(filename);
+
+	ENQUEUE(parquet_file_queue, md5_file_name);
+
+	if (QUEUE_SIZE(parquet_file_queue) > conf->file_count) {
+		remove_old_file();
```

Comment:
One more question. If any of the processes of renaming or calculating md5 failed the file counter will exceed the file_count in configuration.

Reply:
I'm not sure what you mean, if the renaming and calculating md5 fails there will be no enqueue action, the file_count will not grow, and there will be no situation where the file_count exceeds the configured value.

Reply:
Not the length of the queue. I mean the number of files written to disk

Reply:
Got that, already remove file when calculating md5 or renaming failed